src/esp/scene/SceneNode.cpp

```diff
@@ -25,5 +25,48 @@ SceneNode& SceneNode::createChild() {
   return *node;
 }
 
+//! @brief recursively compute the cumulative bounding box of this node's tree.
+const Magnum::Range3D& SceneNode::computeCumulativeBB() {
+  // first copy from your precomputed mesh bb
+  cumulativeBB_ = Magnum::Range3D(meshBB_);
+  auto* child = children().first();
+
+  while (child != nullptr) {
+    // try catch before cast incase a non scene node child is attached here.
+    // Better way?
+    try {
+      SceneNode* child_node = static_cast<SceneNode*>(child);
```

Comment:
You can use `dynamic_cast` here https://en.cppreference.com/w/cpp/language/dynamic_cast -- that'll return a nullptr if the cast is invalid and you can check for that.